src/tribler/core/utilities/pony_utils.py

```diff
@@ -28,6 +35,86 @@
 StatDict = Dict[Optional[str], core.QueryStat]
 
 
+class DatabaseIsCorrupted(Exception):
+    pass
+
+
+def table_exists(cursor: sqlite3.Cursor, table_name: str) -> bool:
+    cursor.execute("SELECT name FROM sqlite_master WHERE type='table' AND name=?", (table_name,))
+    return cursor.fetchone() is not None
+
+
+def get_db_version(db_path, default: int = None) -> int:
+    handle_db_if_corrupted(db_path)
+    version = None
+
+    if db_path.exists():
+        with marking_corrupted_db(db_path):
+            with contextlib.closing(sqlite3.connect(db_path)) as connection:
+                with connection:
+                    cursor = connection.cursor()
+                    if table_exists(cursor, 'MiscData'):
+                        cursor.execute("SELECT value FROM MiscData WHERE name == 'db_version'")
+                        row = cursor.fetchone()
+                        version = int(row[0]) if row else None
+
+    if version is not None:
+        return version
+
+    if default is not None:
+        return default
+
+    raise RuntimeError(f'The version value is not found in database {db_path}')
+
+
+def handle_db_if_corrupted(db_filename: Union[str, Path]):
+    db_path = Path(db_filename)
+    marker_path = _get_corrupted_db_marker_path(db_path)
+    if marker_path.exists():
+        _handle_corrupted_db(db_path)
+
+
+def _handle_corrupted_db(db_path: Path):
+    if db_path.exists():
+        logger.warning(f'Database file was marked as corrupted, removing it: {db_path}')
+        now = datetime.datetime.now().strftime('%Y-%m-%d_%H-%M-%S')
+        db_path.rename(str(db_path) + f'-{now}.corrupted')
```

Comment:
Could you clarify why it is necessary to use {now} timestamp?

Reply:
The database us removed now and not just renamed, so this code was removed